ccm_web/server/src/api/api.service.ts

```diff
@@ -37,6 +37,26 @@ export class APIService {
     }
   }
 
+  async getCourseSections(userLoginId: string, courseId: number): Promise<CanvasCourseSection[] | APIErrorData> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId, '/api/v1/')
+    try {
+      const endpoint = `courses/${courseId}/sections`
+      const queryParams = {'include': ['total_students']} // use list for "include" values
+      logger.debug(`Sending request to Canvas - Endpoint: ${endpoint}; Method: GET`)
+      const response = await requestor.get<CanvasCourseSection[]>(endpoint, queryParams)
```

Comment:
Pagination? Seems to have been requested on the issue. I haven't tried this, but maybe `list` method will make it easy? https://github.com/KTH/canvas-api/blob/master/docs/API.md#listendpoint-queryparams-- 

I started trying this out, seems to work, but I think we need to modify the types for `@kth/canvas-api` (my initial types based on the auto-generated ones seem to be wrong). I'll get back to you, may open a supporting PR to the main repo or to your branch.

```
      const something = requestor.list(endpoint, queryParams)
      console.log(something)
      const allSections = await something.toArray() as CanvasCourseSection[]
      console.log(allSections)
      const sections = allSections.map(s => ({
        id: s.id,
        name: s.name,
        total_students: s.total_students
      }))
```

Canvas page default is 10 I believe, but we can set `per_page` to be something more reasonable, maybe 50? I'd make a global variable for that in `APIService`.

Reply:
Yeah we want to add pagination support to it. Since FE will just ask for section in single call and BE pages throughs all sections and send it. We discussed this in the slack channel as well. 
I think we should set it to `per_page = 100`, since that is the Max we can expect from canvas and for getting section info it should be fast.

Reply:
I might suggest setting per_page to 50% or 100% more than what you might expect to be the max, since it sounds like we're effectively eliminating paging altogether by setting the limit to the expected max.

Setting it higher would have no negative effect on performance and avoid the occasional pagination if there were 101 sections or something.

Reply:
I'm guilty of assuming the pagination was being handled by KTH.  I didn't look into it.  I'll do so.

Any recommendations of Canvas courses that have enough sections to trigger pagination in the response from the API?

Reply:
Keep the `per_page` default, add 11 sections to your course?

Reply:
@lsloan, I opened a PR against your branch with the type changes and I think a work-able pagination solution. I also included the scope change I requested, since I needed it for testing. See https://github.com/lsloan/ccm/pull/1/files

Sorry for maybe stepping on your toes here, I just felt responsible for the wrong types and wanted to fix them. I will create an issue about that this afternoon so that you can link it to your PR (assuming these changes are OK, get merged).

Reply:
OK…  I thought there might be an existing course.

Reply:
As promised, I've created an issue for updating the types (#158), and linked it to this PR.

Reply:
I've accepted @ssciolla's PR to my branch that included code for handling Canvas API pagination.